### ISSUE TITLE
no-empty-blocks doesn't check functions at all

### ISSUE THREAD

**Issue by duaraghav8** [duaraghav8/Ethlint#254]
**Description**
`no-empty-blocks` doesn't report *any* empty functions. In some cases, functions are allowed to be empty, but not all.

**Steps to reproduce**
- A `.soliumrc.json` with `no-empty-blocks` rule enabled.

Code:
```
contract Bar {
  function foo() {
  }
}
```

**Expected behavior**
Rule should report `foo()`.

**Operating System**
Issue is platform-independent.

**Linter version**
`Solium version 1.2.2` - latest at the time of filing issue

**Comments**
Issue originates with this commit -> https://github.com/duaraghav8/Ethlint/commit/4d2f81187352f5fa3585129c7078b74469601e70 which was made because https://github.com/duaraghav8/Ethlint/issues/37.
The issue was valid but the solution was not. It should've been more narrowed instead of completely whitelisting function declarations.

The **fix** is: Allow functions to have empty bodies if they're fallback or `payable`. Allow empty constructors if they're `payable`.

**Priority**
`High`

**Comment by duaraghav8** [comment]
solved in v1.2.3
https://github.com/duaraghav8/Ethlint/blob/master/CHANGELOG.md#123-2019-02-11
